internal/cli/atlas/deployments/delete.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deployments
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/atlas/deployments/options"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/podman"
+	"github.com/mongodb/mongodb-atlas-cli/internal/telemetry"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+var errDeploymentNotFound = errors.New("deployment '%s' not found")
+
+type DeleteOpts struct {
+	cli.OutputOpts
+	cli.GlobalOpts
+	*cli.DeleteOpts
+	options.DeploymentOpts
```

Comment:
I feel this line is a bit odd, what are you reusing?

Reply:
I'm reusing this to get all deployment artifact naming conventions (container, volume, network) in Podman, this will probably be reused by every command to get containers. Trying to use this opts to abstract any logic of how to go from command -> podman (but it's not podman client)

Reply:
but opts are meant to hold flags, are you perhaps trying to reuse something that should belong to podman.Client ?

Reply:
I'm not against a new file but things like hostname, port, etc are tightly coupled with what parameters podman needs

Reply:
we can have a deployment struct

Reply:
we talked offline we will arrange a refactor later